### ISSUE TITLE
let --configFile extend defaultConfig, not replace it

### ISSUE THREAD

**Issue by montmanu** [sitespeedio/sitespeed.io#784]
Currently, if a `--configFile` is provided, the entire config is replaced with it's contents. Because of this, the `configFile` must contain every option required by `sitespeed-io` to run. By allowing the `configFile` to extend (rather than replace) the `defaultConfig`, options for which the `defaultConfig` is suitable can omitted from the `configFile` and the `configFile` can specify only those options which need to override the `defaultConfig`.

**Comment by soulgalore** [comment]
Hi @montmanu thanks for the PR, haven't had time to check it out yet, will do asap.

**Comment by montmanu** [comment]
Thanks, @soulgalore. Any comments or suggestions?

**Comment by soulgalore** [comment]
oops sorry completely forgot about this, let me have a look tomorrow night.

**Comment by montmanu** [comment]
no worries. thanks!

**Comment by abhagupta** [comment]
@soulgalore A  slightly separate issue I found related to using `--configFile` option, that sitespeed.io requires the `resultBaseDir` and a `dataDir` to be a MUST parameters in the config.json file that is passed in.  I drilled down the code and found that there are some checks missing for the above 2 params in `lib/config.js` file. Are these documented to required parameters in case of  `--configFile` option. (I could not find any).     I wanted to send a PR, but seems like this is  not only the problem. After putting checks for these and defaulting to some folder, the tests are complaining about no Urls passed and program running out of heapsize.  (This could be just my machine but just mentioning) .
